core/tauri/src/manager.rs

```diff
@@ -180,20 +181,27 @@ impl<R: Runtime> WindowManager<R> {
     self.inner.menu_ids.clone()
   }
 
-  // setup content for dev-server
+  /// Get the base path to serve data from.
+  ///
+  /// * In dev mode, this will be based on the `devPath` configuration value.
+  /// * Otherwise, this will be based on the `distDir` configuration value.
+  #[cfg(custom_protocol)]
+  fn base_path(&self) -> &AppUrl {
+    &self.inner.config.build.dist_dir
+  }
+
   #[cfg(dev)]
-  fn get_url(&self) -> String {
-    match &self.inner.config.build.dev_path {
-      AppUrl::Url(WindowUrl::External(url)) => url.to_string(),
-      _ => "tauri://localhost".into(),
-    }
+  fn base_path(&self) -> &AppUrl {
+    &self.inner.config.build.dev_path
   }
 
-  #[cfg(custom_protocol)]
-  fn get_url(&self) -> String {
-    match &self.inner.config.build.dist_dir {
-      AppUrl::Url(WindowUrl::External(url)) => url.to_string(),
-      _ => "tauri://localhost".into(),
+  /// Get the base URL to use for webview requests.
+  ///
+  /// In dev mode, this will be based on the `devPath` configuration value.
+  fn get_url(&self) -> Cow<'_, Url> {
+    match self.base_path() {
+      AppUrl::Url(WindowUrl::External(url)) => Cow::Borrowed(url),
+      _ => Cow::Owned(Url::parse("tauri://localhost").unwrap()),
```

Comment:
just a small nit-pick, the comments are not needed, the code is clear enough

Reply:
IDE hinting shows function signature and doc comments, not the internal code, so I like to write doc comments for every function no matter how simple it is, to make it easier when using them elsewhere in the code. So it's mostly a force of habit. And I agree that the code is probably clear enough, but I don't think it's a problem to have these doc comments here?

Reply:
It is not a problem, it was just a preference of mine, but I am not the main maintainer anyway so you'd better wait for his review.